Feat/attribute refactor

SushiRoll53 opened the pull request:
- Introduced PBAttributes and refactored nodes.
- Layout generation services now uses attributes instead of child and children
- Align generation services now uses attributes instead of child and children
- Cleaned and improved interpret code

ivan-015 committed:
Introduced PBAttributes and refactored nodes.

Co-authored-by: Bryan Figueroa <SushiRoll53@users.noreply.github.com>

ivan-015 committed:
Removed service attributes from PBAttribute

ivan-015 committed:
Add null check to child setter of IntermediateNode

ivan-015 committed:
Merge branch 'dev' into feat/attribute-refactor

Commit:
Fix on Appbar, missing commas and widget

Commit:
Fixed comma issue

ivan-015 committed:
Merge branch 'feat/attribute-refactor' of https://github.com/Parabeac/Parabeac-Core into feat/attribute-refactor

Commit:
Merge branch 'dev' into feat/attribute-refactor

Commit:
Removed interpreter inside app bar

Commit:
Null safety for alignChild

Commit:
Layout generation services now uses attributes instead of child and children

Commit:
Align generation service now uses attributes instead of child and children

Commit:
Removed methods no longer needed

Commit:
Cleaned and improved interpret code

Commit:
Added custom aligment for Appbar title

Commit:
Added support for crossAxisAlignment and mainAxisAlignment for row and column

ivan-015 commented:
Are we still putting the cross and main axis alignment fix on hold? 🤔

ivan-015 committed:
Add null check when traversing through attributes.
Added attributes to Layout Generation Unit Test

siliconivan reviewed (approved):
Since this is not going into master, going to approve.

Commit:
Fixed conflicts

ivan-015 committed:
Removed adding attribute to tempGroup.
This is done because the when() method already handles calling tempGroup's attributes.

ivan-015 reviewed (approved):
LGTM
